ext/scheduler/airflow2/client.go

```diff
@@ -0,0 +1,166 @@
+	var resp []byte
+	host, authToken, err := ac.getHostAuth(projectSpec)
+	if err != nil {
+		return resp, err
+	}
+	request, err := http.NewRequestWithContext(ctx, r.method, ac.buildEndPoint(host, r.URL, r.param), bytes.NewBuffer(r.body))
+	if err != nil {
+		return resp, fmt.Errorf("failed to build http request for %s due to %w", r.URL, err)
+	}
+	request.Header.Set("Content-Type", "application/json")
+	request.Header.Set("Authorization", fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte(authToken))))
+
+	HTTPResp, respErr := ac.client.Do(request)
+	if respErr != nil {
+		return resp, fmt.Errorf("failed to call airflow %s due to %w", r.URL, respErr)
+	}
+	if HTTPResp.StatusCode != http.StatusOK {
+		HTTPResp.Body.Close()
+		return resp, fmt.Errorf("status code received %d on calling %s", HTTPResp.StatusCode, r.URL)
+	}
+	return ac.parseResponse(HTTPResp)
+}
+
+func (ac airflowClient) getHostAuth(projectSpec models.ProjectSpec) (string, string, error) {
+	schdHost, ok := projectSpec.Config[models.ProjectSchedulerHost]
+	if !ok {
+		return "", "", fmt.Errorf("scheduler host not set for %s", projectSpec.Name)
+	}
+	authToken, ok := projectSpec.Secret.GetByName(models.ProjectSchedulerAuth)
+	if !ok {
+		return "", "", fmt.Errorf("%s secret not configured for project %s", models.ProjectSchedulerAuth, projectSpec.Name)
+	}
+	schdHost = strings.ReplaceAll(schdHost, "http://", "")
+	return schdHost, authToken, nil
+}
+
+func (ac airflowClient) parseResponse(resp *http.Response) ([]byte, error) {
+	var body []byte
+	body, err := io.ReadAll(resp.Body)
+	resp.Body.Close()
+	if err != nil {
+		return body, fmt.Errorf("failed to read airflow response: %w", err)
+	}
+	return body, nil
+}
+
+func (ac airflowClient) buildEndPoint(host string, URL string, pathParam string) string {
+	host = strings.Trim(host, "/")
+	u := &url.URL{
+		Scheme: "http",
+		Host:   host,
+	}
+	if pathParam != "" {
+		u.Path = "/" + strings.ReplaceAll(URL, "%s", pathParam)
+	} else {
+		u.Path = "/" + URL
+	}
+	return u.String()
+}
+
+func toJobStatus(list DagRunList) ([]models.JobStatus, error) {
+	var jobStatus []models.JobStatus
+	for _, status := range list.DagRuns {
+		jobStatus = append(jobStatus, models.JobStatus{
+			ScheduledAt: status.ExecutionDate,
+			State:       models.JobRunState(status.State),
+		})
+	}
+	return jobStatus, nil
+}
+
+func getDagRunReqBody(param *models.JobQuery) DagRunReqBody {
+	if param.OnlyLastRun {
+		return DagRunReqBody{
+			OrderBy:    "-execution_date",
+			PageOffset: 0,
+			PageLimit:  1,
+			DagIds:     []string{param.Name},
+		}
+	}
+	return DagRunReqBody{
+		OrderBy:          "execution_date",
+		PageOffset:       0,
+		PageLimit:        Hundred,
+		DagIds:           []string{param.Name},
+		ExecutionDateGte: param.StartDate.Format(airflowDateFormat),
+		ExecutionDateLte: param.EndDate.Format(airflowDateFormat),
+	}
+}
+
+func getJobRunList(res DagRunList) []models.JobRun {
```

Comment:
function name & the return type are missleading

Reply:
done in latest commit